### PR TITLE
feat: add onOpenChange to popover

### PR DESCRIPTION
**Pull request by marshallpete**
## Description:

Add `onOpenChange` to `ChartPopover`. This will allow users to trigger side effects when the chartpopover opens of closes.

## Story:
ChartPopover > OnOpenChange

## Screenshots (if appropriate):

https://github.com/adobe/react-spectrum-charts/assets/40001449/3a0b4a1c-afd6-40c8-963e-397fd6c85bf0

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/RscChart.tsx [hunk 1]
```diff
@@ -288,41 +287,45 @@ export const RscChart = forwardRef<ChartHandle, RscChartProps>(
...
-				/>
+				{popovers.map((popover) => (
+					<ChartDialog
+						key={popover.key}
+						datum={selectedData.current}
+						targetElement={popoverAnchorRef}
+						setIsPopoverOpen={setIsPopoverOpen}
+						popover={popover}
+					/>
+				))}
```
Comment (marshallpete) [inline src/RscChart.tsx:298]: Better practice for react

File: src/RscChart.tsx [hunk 2]
```diff
@@ -100,7 +99,7 @@ export const RscChart = forwardRef<ChartHandle, RscChartProps>(
 		const selectedDataBounds = useRef<MarkBounds>();
 		const popoverAnchorRef = useRef<HTMLDivElement>(null);
 
-		const [popoverIsOpen, setPopoverIsOpen] = useState<boolean>(false); // tracks the open/close state of the popover
+		const [isPopoverOpen, setIsPopoverOpen] = useState<boolean>(false); // tracks the open/close state of the popover
```
Comment (marshallpete) [inline src/RscChart.tsx:102]: rename so it's more clear

File: src/RscChart.tsx [hunk 3]
```diff
@@ -288,41 +287,45 @@ export const RscChart = forwardRef<ChartHandle, RscChartProps>(
...
 	return (
 		<DialogTrigger
 			type="popover"
 			mobileType="tray"
 			targetRef={targetElement}
-			onOpenChange={setPopoverState}
+			onOpenChange={(isOpen) => {
+				onOpenChange?.(isOpen);
+				setIsPopoverOpen(isOpen);
+			}}
```
Comment (marshallpete) [inline src/RscChart.tsx:318]: support onOpenChange

File: src/types/Chart.ts [hunk 4]
```diff
@@ -559,6 +559,8 @@ export interface ChartPopoverProps {
 	minHeight?: number;
 	/** Maximum height of the popover */
 	maxHeight?: number;
+	/** handler that is called when the popover's open state changes */
```
Comment (ToyWalrus) [inline src/types/Chart.ts:562]: +1 for doc comments 👍

**Review by ToyWalrus — APPROVED** [review]
Looks great

**Pull request merged**